quickwit-janitor/src/actors/janitor_service.rs

```diff
@@ -73,32 +109,89 @@ impl Actor for JanitorService {
         &mut self,
         ctx: &ActorContext<Self>,
     ) -> Result<(), quickwit_actors::ActorExitStatus> {
-        self.handle(Loop, ctx).await
+        self.handle(SpawnGarbageCollector, ctx).await?;
+        self.handle(Observe, ctx).await?;
+        self.handle(Supervise, ctx).await
+    }
+}
+
+#[async_trait]
+impl Handler<Supervise> for JanitorService {
+    type Reply = ();
+
+    async fn handle(
+        &mut self,
+        _message: Supervise,
+        ctx: &ActorContext<Self>,
+    ) -> Result<(), ActorExitStatus> {
+        if let Some(gc_handle) = &self.garbage_collector_handle_opt {
```

Comment:
What should we do if the GC actor is unhealthy?
In which cases the GC is unhealthy?

Reply:
As a first PR, I would even remove the supervise stuff and see what we will do the retention policy actors.

Reply:
My reasoning is that when we fail to GC an index this actor will be in the wrong state. 
But we could just log the error and let the service keep running in which case we don't need this supervision.